### ISSUE TITLE
Avoid cluttering the home directory

### ISSUE THREAD

**Issue by klmr** [randy3k/radian#202]
Currently the history file is unconditionally written to `$HOME/.radian_history`. Ideally this should be configurable, but at the very least the default location should follow [XDG base dir conventions](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) (i.e. default to`$HOME/.local/share/radian/history`).

(On macOS there’s an alternative convention to use `$HOME/Library/Application Support` but from experience applications are split 50/50 about whether they use that rather than XDG, and both are fine.)

**Comment by randy3k** [comment]
closed by 5546b07

However, it requires you to specify the setting `radian.history_path` in `~/.Rprofile` or `~/.radian_profile`. 

It may also make sense to allow custom profile location. In R, `R_PROFILE_USER ` is used to control the path to the profile, but I really dislike the idea of using an environmental variable.

**Comment by klmr** [comment]
This solution is fine. I still feel that the default should do The Right Thing™️ but as long as this is configurable that is secondary.

I’m not sure whether it makes sense to specify the history path as a *folder* rather than the actual file.

**Comment by randy3k** [comment]
The reason was that we still need to support local history. i.e., if `.radian_history` exists in the pwd, we would use it instead of the global history.

If we allow user to specify things like `~/.radian/history`, do you still read the local history? If yes, which file to read? `history` or `.radian_history`. It just seems to create more confusing.


Perhaps `radian.history_location` is a better setting name.

**Comment by klmr** [comment]
I see these two things as completely orthogonal. For reference, that’s also how e.g. git handles global and local configuration, or global and local ignore files.

The local file always has the same name (`.radian_history` is fine here).

— Incidentally, R itself does this badly: you can’t have a custom user profile configuration in addition to directory-local configuration (because `R_PROFILE_USER` overrides that).

**Comment by randy3k** [comment]
Future updated.

```r
# where the global history is stored, environmental variables will be expanded
# note that "~" is expanded to %USERPROFILE% in Windows
options(radian.global_history_file = "~/.radian_history")
# the filename that local history is stored, this file would be used instead of
# `radian.global_history_file` if it exists in the current working directory
options(radian.local_history_file = ".radian_history")
```



> — Incidentally, R itself does this badly: you can’t have a custom user profile configuration in addition to directory-local configuration (because `R_PROFILE_USER` overrides that).

Ahh, good point. `radian` should load both global and local profiles if they both exist.

**Comment by klmr** [comment]
> `radian` should load both global and local profiles if they both exist.

That would be inconsistent with R, see [`?Startup`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/Startup.html):

> unless `--no-init-file` was given, R searches for a user profile …. The path of this file can be specified by the `R_PROFILE_USER` environment variable (and tilde expansion will be performed). If this is unset, a file called ‘`.Rprofile`’ is searched for in the current directory or in the user's home directory (in that order).

In other words, R only loads *one* user profile file (in addition to the *site-wide* profile), and path-local profiles override `~/.Rprofile`.

**Comment by klmr** [comment]
Damn, one more nitpick:

Your recent changes in d34a957a0e140c81dfa41da105d2c6b888095283 use `XDG_DATA_HOME` to store the configuration (`profile`). However, configuration should go into `XDG_CONFIG_HOME` (default `~/.config`) instead.

**Comment by randy3k** [comment]
My bad, I am not very familiar with the XDG things, fixed 453e3ff.
